oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyState.java

```diff
@@ -38,24 +45,64 @@
 import org.apache.jackrabbit.oak.plugins.memory.StringPropertyState;
 import org.apache.jackrabbit.oak.plugins.value.Conversions;
 import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * PropertyState implementation with lazy parsing of the JSOP encoded value.
  */
 final class DocumentPropertyState implements PropertyState {
 
+    private static final Logger LOG = LoggerFactory.getLogger(DocumentPropertyState.class);
+
     private final DocumentNodeStore store;
 
     private final String name;
 
     private final String value;
 
     private PropertyState parsed;
+    private final byte[] compressedValue;
+    private final Compression compression;
+
+    private static final int DEFAULT_COMPRESSION_THRESHOLD = SystemPropertySupplier
+            .create("oak.documentMK.stringCompressionThreshold ", -1).loggingTo(LOG).get();
```

Comment:
I would suggest to refactor this. 
One aspect is that this variable shouldn't be called "default", as it is actually the real configured value - the default is always -1, that can be a separate, hard-code const.
A second aspect is how this variable can be controlled from tests. This has been a controversial topic before - in my opinion if we like to test different values of say "compressionThreshold", then there should be a clean way to achieve that. A `final` in my view isn't ideal, as compilers can in theory inline final variables if they so wish. What I think can be an option is to add an explicit (static) setter method that is only used for testing - and that allows overwriting this field explicitly. With this, the need for explicit test order plus the various reflections in the junit class disappear.

Reply:
Done